[PDI-15258] Step fails when no input received

matthewtckr opened the pull request.

wingman-pentaho commented:
Build completed
===============

:x: This pull request has errors. They will need to be addressed before it can be accepted. See below for more details. Some links are also available below for further assistance in addressing those issues.

Build Commands:
------
1. ant -Dtestreports.dir=bin/reports/unit-test -f plugins/kettle-json-plugin/build.xml clean-all resolve jacoco && ant -Dtestreports.dir=bin/reports/integration-test -f plugins/kettle-json-plugin/build.xml jacoco-integration checkstyle publish-local

Cleanup Commands:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  plugins/kettle-json-plugin/src/org/pentaho/di/trans/steps/jsoninput/JsonInput.java
  plugins/kettle-json-plugin/src/org/pentaho/di/trans/steps/jsoninput/JsonInputData.java
  plugins/kettle-json-plugin/src/org/pentaho/di/trans/steps/jsoninput/JsonInputMeta.java
  plugins/kettle-json-plugin/test-src/org/pentaho/di/trans/steps/jsoninput/JsonInputMetaLoadSaveTest.java

Newly Broken Tests:
======================
org.pentaho.di.trans.steps.jsoninput.JsonInputTest.testNullFileList:
-----------------------------------------------
junit.framework.AssertionFailedError
```
junit.framework.AssertionFailedError: expected:<2> but was:<0>
	at org.pentaho.di.trans.steps.jsoninput.JsonInputTest.testNullFileList(JsonInputTest.java:620)

```

Unit test coverage change:
============
org.pentaho.di.trans.steps.jsoninput.JsonInput:
 * Branch Change : **-5.03%**
 * Complexity Change : **-6.16%**
 * Instruction Change : **-3.99%**
 * Line Change : **-3.58%**

org.pentaho.di.trans.steps.jsoninput.JsonInputField:
 * Branch Change :  + 16.67%
 * Complexity Change :  + 25.00%
 * Instruction Change :  + 70.76%
 * Line Change :  + 54.39%
 * Method Change :  + 30.77%

org.pentaho.di.trans.steps.jsoninput.JsonInputMeta:
 * Branch Change :  + 18.18%
 * Complexity Change :  + 12.17%
 * Instruction Change :  + 42.68%
 * Line Change :  + 36.09%
 * Method Change :  + 9.76%

org.pentaho.di.trans.steps.jsoninput.JsonInputMeta.InputFiles:
 * Branch Change :  + 100.00%
 * Complexity Change :  + 50.00%
 * Instruction Change :  + 40.00%
 * Line Change :  + 27.78%
 * Method Change :  + 33.33%

wingman-pentaho commented:
Base build failed with return code 1

wingman-pentaho commented:
Base build failed with return code 1

wingman-pentaho commented:
Base build failed with return code 1

wingman-pentaho commented:
Base build failed with return code 1

wingman-pentaho commented:
Base build failed with return code 1

wingman-pentaho commented:
Build completed
===============

:x: This pull request has errors. They will need to be addressed before it can be accepted. See below for more details. Some links are also available below for further assistance in addressing those issues.

Build Commands:
------
1. ant -Dtestreports.dir=bin/reports/unit-test -f engine/build.xml clean-all resolve jacoco && ant -Dtestreports.dir=bin/reports/integration-test -f engine/build.xml jacoco-integration checkstyle publish-local
2. ant -Dtestreports.dir=bin/reports/unit-test -f plugins/kettle-json-plugin/build.xml clean-all resolve jacoco && ant -Dtestreports.dir=bin/reports/integration-test -f plugins/kettle-json-plugin/build.xml jacoco-integration checkstyle publish-local

Cleanup Commands:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  engine/test-src/org/pentaho/di/trans/RowStepCollector.java
  engine/test-src/org/pentaho/di/trans/TransTestFactory.java
  plugins/kettle-json-plugin/src/org/pentaho/di/trans/steps/jsoninput/JsonInput.java
  plugins/kettle-json-plugin/src/org/pentaho/di/trans/steps/jsoninput/JsonInputData.java
  plugins/kettle-json-plugin/src/org/pentaho/di/trans/steps/jsoninput/JsonInputMeta.java
  plugins/kettle-json-plugin/test-src/org/pentaho/di/trans/steps/jsoninput/JsonInputIntTest.java
  plugins/kettle-json-plugin/test-src/org/pentaho/di/trans/steps/jsoninput/JsonInputMetaLoadSaveTest.java

Newly Broken Tests:
======================
org.pentaho.di.trans.steps.jsoninput.JsonInputIntTest.testNoInput:
-----------------------------------------------
org.pentaho.di.core.exception.KettleException
```
org.pentaho.di.core.exception.KettleException: 
We failed to initialize at least one step.  Execution can not begin!


	at org.pentaho.di.trans.Trans.prepareExecution(Trans.java:1142)
	at org.pentaho.di.trans.TransTestFactory.executeTestTransformation(TransTestFactory.java:161)
	at org.pentaho.di.trans.TransTestFactory.executeTestTransformation(TransTestFactory.java:142)
	at org.pentaho.di.trans.steps.jsoninput.JsonInputIntTest.testNoInput(JsonInputIntTest.java:56)

```


Still Broken Tests:
======================
org.pentaho.di.trans.steps.rest.RestIT.testNoContent:
-----------------------------------------------
java.lang.ArrayIndexOutOfBoundsException
```
java.lang.ArrayIndexOutOfBoundsException: 11264
	at org.objectweb.asm.ClassReader.readClass(Unknown Source)
	at org.objectweb.asm.ClassReader.accept(Unknown Source)
	at org.objectweb.asm.ClassReader.accept(Unknown Source)
	at com.sun.jersey.spi.scanning.AnnotationScannerListener.onProcess(AnnotationScannerListener.java:136)
	at com.sun.jersey.core.spi.scanning.JarFileScanner.scan(JarFileScanner.java:97)
	at com.sun.jersey.core.spi.scanning.JarFileScanner$1.f(JarFileScanner.java:74)
	at com.sun.j
```
org.pentaho.di.trans.steps.rest.RestIT.testResponseHeader:
-----------------------------------------------
java.lang.ArrayIndexOutOfBoundsException
```
java.lang.ArrayIndexOutOfBoundsException

```

Unit test coverage change:
============
org.pentaho.di.job.entries.evalfilesmetrics.JobEntryEvalFilesMetrics:
 * Branch Change : **-.39%**
 * Complexity Change : **-.51%**
 * Instruction Change : **-.08%**

org.pentaho.di.job.entries.sql.JobEntrySQL:
 * Instruction Change : **-3.28%**
 * Line Change : **-3.87%**

org.pentaho.di.trans.steps.calculator.CalculatorMetaFunction:
 * Branch Change :  + 8.70%
 * Complexity Change :  + 3.08%
 * Instruction Change :  + .75%
 * Line Change :  + .90%

org.pentaho.di.trans.steps.excelwriter.ExcelWriterStepMeta:
 * Branch Change : **-1.28%**
 * Complexity Change : **-.79%**
 * Instruction Change : **-.04%**

org.pentaho.di.trans.steps.getfilesrowscount.GetFilesRowsCountMeta:
 * Branch Change : **-1.85%**
 * Complexity Change : **-1.30%**

org.pentaho.di.trans.steps.ifnull.IfNullMeta:
 * Branch Change : **-2.38%**
 * Complexity Change : **-1.64%**
 * Instruction Change : **-.17%**

org.pentaho.di.trans.steps.pgbulkloader.PGBulkLoaderMeta:
 * Branch Change : **-.88%**
 * Complexity Change : **-.93%**

org.pentaho.di.trans.steps.setvalueconstant.SetValueConstantMeta:
 * Instruction Change : **-.14%**

org.pentaho.di.trans.steps.jsoninput.JsonInput:
 * Branch Change : **-.08%**
 * Complexity Change : **-.74%**
 * Instruction Change : **-.08%**
 * Line Change : **-.88%**

org.pentaho.di.trans.steps.jsoninput.JsonInputField:
 * Branch Change :  + 16.67%
 * Complexity Change :  + 25.00%
 * Instruction Change :  + 70.76%
 * Line Change :  + 54.39%
 * Method Change :  + 30.77%

org.pentaho.di.trans.steps.jsoninput.JsonInputMeta:
 * Branch Change :  + 18.18%
 * Complexity Change :  + 14.78%
 * Instruction Change :  + 43.41%
 * Line Change :  + 36.83%
 * Method Change :  + 13.41%

org.pentaho.di.trans.steps.jsoninput.JsonInputMeta.InputFiles:
 * Branch Change :  + 100.00%
 * Complexity Change :  + 50.00%
 * Instruction Change :  + 40.00%
 * Line Change :  + 27.78%
 * Method Change :  + 33.33%

Integration test coverage change:
============
org.pentaho.di.trans.step.BaseStep:
 * Branch Change : **-.29%**
 * Complexity Change : **-.18%**
 * Instruction Change : **-.13%**
 * Line Change : **-.08%**

License header violations:
==========================
1. Copyright year is not the current year on engine/test-src/org/pentaho/di/trans/RowStepCollector.java. Found 2013. Was expecting 2016

matthewtckr commented:
Base build failed with return code 127

matthewtckr commented:
Base build failed with return code 127

matthewtckr commented:
Base build failed with return code 127

matthewtckr commented:
Build completed
===============

:x: This pull request has errors. They will need to be addressed before it can be accepted. See below for more details. Some links are also available below for further assistance in addressing those issues.

Build Commands:
------
1. ant -Dtestreports.dir=bin/reports/unit-test -f engine/build.xml clean-all resolve jacoco && ant -Dtestreports.dir=bin/reports/integration-test -f engine/build.xml jacoco-integration checkstyle publish-local
2. ant -Dtestreports.dir=bin/reports/unit-test -f plugins/kettle-json-plugin/build.xml clean-all resolve jacoco && ant -Dtestreports.dir=bin/reports/integration-test -f plugins/kettle-json-plugin/build.xml jacoco-integration checkstyle publish-local

Cleanup Commands:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  engine/test-src/org/pentaho/di/trans/RowStepCollector.java
  engine/test-src/org/pentaho/di/trans/TransTestFactory.java
  plugins/kettle-json-plugin/src/org/pentaho/di/trans/steps/jsoninput/JsonInput.java
  plugins/kettle-json-plugin/src/org/pentaho/di/trans/steps/jsoninput/JsonInputData.java
  plugins/kettle-json-plugin/src/org/pentaho/di/trans/steps/jsoninput/JsonInputMeta.java
  plugins/kettle-json-plugin/src/org/pentaho/di/trans/steps/jsoninput/JsonReader.java
  plugins/kettle-json-plugin/src/org/pentaho/di/trans/steps/jsoninput/NJSONArray.java
  plugins/kettle-json-plugin/test-src/org/pentaho/di/trans/steps/jsoninput/JsonInputIntTest.java
  plugins/kettle-json-plugin/test-src/org/pentaho/di/trans/steps/jsoninput/JsonInputMetaLoadSaveTest.java

Newly Broken Tests:
======================
org.pentaho.di.trans.steps.monetdbbulkloader.MonetDBBulkLoaderMetaTest.testSerialization:
-----------------------------------------------
java.lang.RuntimeException
```
java.lang.RuntimeException: Error validating autoSchema
	at org.pentaho.di.trans.steps.loadsave.LoadSaveTester.validateLoadedMeta(LoadSaveTester.java:193)
	at org.pentaho.di.trans.steps.loadsave.LoadSaveTester.testXmlRoundTrip(LoadSaveTester.java:238)
	at org.pentaho.di.trans.steps.loadsave.LoadSaveTester.testSerialization(LoadSaveTester.java:199)
	at org.pentaho.di.trans.steps.monetdbbulkloader.MonetDBBulkLoaderMetaTest.testSerialization(MonetDBBulkLoaderMetaTest.java:133)
Caused by: org.pentah
```


Still Broken Tests:
======================
org.pentaho.di.trans.steps.rest.RestIT.testNoContent:
-----------------------------------------------
java.lang.ArrayIndexOutOfBoundsException
```
java.lang.ArrayIndexOutOfBoundsException: 11264
	at org.objectweb.asm.ClassReader.readClass(Unknown Source)
	at org.objectweb.asm.ClassReader.accept(Unknown Source)
	at org.objectweb.asm.ClassReader.accept(Unknown Source)
	at com.sun.jersey.spi.scanning.AnnotationScannerListener.onProcess(AnnotationScannerListener.java:136)
	at com.sun.jersey.core.spi.scanning.JarFileScanner.scan(JarFileScanner.java:97)
	at com.sun.jersey.core.spi.scanning.JarFileScanner$1.f(JarFileScanner.java:74)
	at com.sun.j
```
org.pentaho.di.trans.steps.rest.RestIT.testResponseHeader:
-----------------------------------------------
java.lang.ArrayIndexOutOfBoundsException
```
java.lang.ArrayIndexOutOfBoundsException

```

Unit test coverage change:
============
org.pentaho.di.core.injection.bean.BeanInjector:
 * Branch Change :  + 1.27%
 * Complexity Change :  + 2.00%

org.pentaho.di.job.entries.waitforsql.JobEntryWaitForSQL:
 * Branch Change : **-.67%**
 * Complexity Change : **-.94%**
 * Instruction Change : **-.13%**

org.pentaho.di.trans.steps.checksum.CheckSumMeta:
 * Branch Change :  + 2.94%
 * Complexity Change :  + 3.08%
 * Instruction Change :  + .70%
 * Line Change :  + .57%

org.pentaho.di.trans.steps.combinationlookup.CombinationLookupMeta:
 * Instruction Change : **-.04%**

org.pentaho.di.trans.steps.dbproc.DBProcMeta:
 * Instruction Change :  + .11%

org.pentaho.di.trans.steps.dimensionlookup.DimensionLookupMeta:
 * Branch Change :  + .66%
 * Complexity Change :  + .41%
 * Instruction Change :  + .50%
 * Line Change :  + .24%

org.pentaho.di.trans.steps.groupby.GroupByMeta:
 * Branch Change : **-3.12%**
 * Complexity Change : **-3.16%**
 * Instruction Change : **-.14%**
 * Line Change : **-.37%**

org.pentaho.di.trans.steps.ldapoutput.LDAPOutputMeta:
 * Instruction Change : **-.06%**

org.pentaho.di.trans.steps.memgroupby.MemoryGroupByMeta:
 * Branch Change : **-1.25%**
 * Complexity Change : **-1.41%**
 * Instruction Change : **-.18%**
 * Line Change : **-.51%**

org.pentaho.di.trans.steps.monetdbbulkloader.MonetDBBulkLoaderMeta:
 * Branch Change : **-6.85%**
 * Complexity Change : **-4.38%**
 * Instruction Change : **-14.09%**
 * Line Change : **-11.53%**
 * Method Change : **-6.25%**

org.pentaho.di.trans.steps.pgbulkloader.PGBulkLoaderMeta:
 * Branch Change : **-1.75%**
 * Complexity Change : **-.93%**
 * Instruction Change : **-.23%**
 * Line Change : **-.28%**

org.pentaho.di.trans.steps.uniquerows.UniqueRowsMeta:
 * Instruction Change : **-.18%**

org.pentaho.di.trans.steps.jsoninput.JsonInput:
 * Branch Change :  + 1.21%
 * Complexity Change :  + 1.24%
 * Instruction Change :  + .52%
 * Line Change :  + .48%

org.pentaho.di.trans.steps.jsoninput.JsonInputField:
 * Branch Change :  + 16.67%
 * Complexity Change :  + 25.00%
 * Instruction Change :  + 70.76%
 * Line Change :  + 54.39%
 * Method Change :  + 30.77%

org.pentaho.di.trans.steps.jsoninput.JsonInputMeta:
 * Branch Change :  + 18.18%
 * Complexity Change :  + 14.78%
 * Instruction Change :  + 43.41%
 * Line Change :  + 36.83%
 * Method Change :  + 13.41%

org.pentaho.di.trans.steps.jsoninput.JsonInputMeta.InputFiles:
 * Branch Change :  + 100.00%
 * Complexity Change :  + 50.00%
 * Instruction Change :  + 40.00%
 * Line Change :  + 27.78%
 * Method Change :  + 33.33%

matthewtckr commented:
Base build failed with return code 1

matthewtckr commented:
Base build failed with return code 1

matthewtckr commented:
Base build failed with return code 1

matthewtckr commented:
Head build failed with return code 1

matthewtckr commented:
Build completed
===============

:fire: This pull request has some issues. It would be preferable to fix them in order for it to be just perfect. See below for more details. Some links are also available below for further assistance in addressing those issues.

Build Commands:
------
1. ant -Dtestreports.dir=bin/reports/unit-test -f engine/build.xml clean-all resolve jacoco && ant -Dtestreports.dir=bin/reports/integration-test -f engine/build.xml jacoco-integration checkstyle publish-local
2. ant -Dtestreports.dir=bin/reports/unit-test -f plugins/kettle-json-plugin/build.xml clean-all resolve jacoco && ant -Dtestreports.dir=bin/reports/integration-test -f plugins/kettle-json-plugin/build.xml jacoco-integration checkstyle publish-local

Cleanup Commands:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  engine/test-src/org/pentaho/di/trans/RowStepCollector.java
  engine/test-src/org/pentaho/di/trans/TransTestFactory.java
  plugins/kettle-json-plugin/src/org/pentaho/di/trans/steps/jsoninput/JsonInput.java
  plugins/kettle-json-plugin/src/org/pentaho/di/trans/steps/jsoninput/JsonInputData.java
  plugins/kettle-json-plugin/src/org/pentaho/di/trans/steps/jsoninput/JsonInputMeta.java
  plugins/kettle-json-plugin/src/org/pentaho/di/trans/steps/jsoninput/JsonReader.java
  plugins/kettle-json-plugin/src/org/pentaho/di/trans/steps/jsoninput/NJSONArray.java
  plugins/kettle-json-plugin/test-src/org/pentaho/di/trans/steps/jsoninput/JsonInputIntTest.java
  plugins/kettle-json-plugin/test-src/org/pentaho/di/trans/steps/jsoninput/JsonInputMetaLoadSaveTest.java

Newly Fixed Tests:
======================
org.pentaho.di.trans.steps.textfileinput.TextFileInputMetaLoadSaveTest.testSerialization:
-----------------------------------------------
java.lang.RuntimeException
```
java.lang.RuntimeException: Error validating dateFormatLocale
	at org.pentaho.di.trans.steps.loadsave.LoadSaveTester.validateLoadedMeta(LoadSaveTester.java:174)
	at org.pentaho.di.trans.steps.loadsave.LoadSaveTester.testXmlRoundTrip(LoadSaveTester.java:221)
	at org.pentaho.di.trans.steps.loadsave.LoadSaveTester.testSerialization(LoadSaveTester.java:180)
	at org.pentaho.di.trans.steps.textfileinput.TextFileInputMetaLoadSaveTest.testSerialization(TextFileInputMetaLoadSaveTest.java:157)
Caused by: 
```

Still Broken Tests:
======================
org.pentaho.di.trans.steps.monetdbbulkloader.MonetDBBulkLoaderMetaTest.testSerialization:
-----------------------------------------------
java.lang.RuntimeException
```
java.lang.RuntimeException: Error validating autoSchema
	at org.pentaho.di.trans.steps.loadsave.LoadSaveTester.validateLoadedMeta(LoadSaveTester.java:174)
	at org.pentaho.di.trans.steps.loadsave.LoadSaveTester.testXmlRoundTrip(LoadSaveTester.java:221)
	at org.pentaho.di.trans.steps.loadsave.LoadSaveTester.testSerialization(LoadSaveTester.java:180)
	at org.pentaho.di.trans.steps.monetdbbulkloader.MonetDBBulkLoaderMetaTest.testSerialization(MonetDBBulkLoaderMetaTest.java:133)
Caused by: org.pentah
```
org.pentaho.di.trans.steps.rest.RestIT.testNoContent:
-----------------------------------------------
java.lang.ArrayIndexOutOfBoundsException
```
java.lang.ArrayIndexOutOfBoundsException: 11264
	at org.objectweb.asm.ClassReader.readClass(Unknown Source)
	at org.objectweb.asm.ClassReader.accept(Unknown Source)
	at org.objectweb.asm.ClassReader.accept(Unknown Source)
	at com.sun.jersey.spi.scanning.AnnotationScannerListener.onProcess(AnnotationScannerListener.java:136)
	at com.sun.jersey.core.spi.scanning.JarFileScanner.scan(JarFileScanner.java:97)
	at com.sun.jersey.core.spi.scanning.JarFileScanner$1.f(JarFileScanner.java:74)
	at com.sun.j
```
org.pentaho.di.trans.steps.rest.RestIT.testResponseHeader:
-----------------------------------------------
java.lang.ArrayIndexOutOfBoundsException
```
java.lang.ArrayIndexOutOfBoundsException

```

Unit test coverage change:
============
org.pentaho.di.core.auth.core.impl.DefaultAuthenticationConsumerFactory:
 * Branch Change : **-7.14%**
 * Complexity Change : **-7.69%**

org.pentaho.di.job.entries.evalfilesmetrics.JobEntryEvalFilesMetrics:
 * Branch Change : **-.39%**
 * Complexity Change : **-.51%**
 * Instruction Change : **-.08%**

org.pentaho.di.job.entries.sql.JobEntrySQL:
 * Instruction Change : **-3.28%**
 * Line Change : **-3.87%**

org.pentaho.di.job.entries.waitforsql.JobEntryWaitForSQL:
 * Branch Change : **-.67%**
 * Complexity Change : **-.94%**
 * Instruction Change : **-.13%**

org.pentaho.di.trans.steps.checksum.CheckSumMeta:
 * Instruction Change : **-.11%**

org.pentaho.di.trans.steps.datagrid.DataGridMeta:
 * Instruction Change :  + .09%

org.pentaho.di.trans.steps.dbproc.DBProcMeta:
 * Instruction Change :  + .11%

org.pentaho.di.trans.steps.groupby.GroupByMeta:
 * Branch Change :  + 3.12%
 * Complexity Change :  + 3.16%
 * Instruction Change :  + .14%
 * Line Change :  + .37%

org.pentaho.di.trans.steps.ifnull.IfNullMeta:
 * Instruction Change : **-.08%**

org.pentaho.di.trans.steps.ldapoutput.LDAPOutputMeta:
 * Instruction Change :  + .06%

org.pentaho.di.trans.steps.memgroupby.MemoryGroupByMeta:
 * Branch Change : **-1.25%**
 * Complexity Change : **-1.41%**

org.pentaho.di.trans.steps.orabulkloader.OraBulkLoaderMeta:
 * Branch Change :  + 2.38%
 * Complexity Change :  + .83%
 * Instruction Change :  + .19%
 * Line Change :  + .22%

org.pentaho.di.trans.steps.replacestring.ReplaceStringMeta:
 * Instruction Change :  + .07%

org.pentaho.di.trans.steps.setvalueconstant.SetValueConstantMeta:
 * Instruction Change :  + .14%

org.pentaho.di.trans.steps.singlethreader.SingleThreaderMeta:
 * Instruction Change :  + .08%

org.pentaho.di.trans.steps.symmetriccrypto.symmetriccryptotrans.SymmetricCryptoTransMeta:
 * Branch Change :  + 3.33%
 * Complexity Change :  + 1.85%
 * Instruction Change :  + .28%

org.pentaho.di.trans.steps.textfileinput.TextFileFilter:
 * Complexity Change :  + 9.09%
 * Instruction Change :  + 9.26%
 * Line Change :  + 8.33%
 * Method Change :  + 9.09%

org.pentaho.di.trans.steps.textfileinput.TextFileInputMeta:
 * Branch Change :  + 12.23%
 * Complexity Change :  + 5.20%
 * Instruction Change :  + 30.18%
 * Line Change :  + 22.74%
 * Method Change :  + 3.21%

org.pentaho.di.trans.steps.webservices.WebServiceMeta:
 * Branch Change : **-1.79%**
 * Complexity Change : **-1.19%**
 * Instruction Change : **-.07%**

Integration test coverage change:
============
org.pentaho.di.trans.step.BaseStep:
 * Branch Change :  + .29%
 * Complexity Change :  + .18%
 * Instruction Change :  + .13%
 * Line Change :  + .08%
